lib/watch-sidebar.js

```diff
@@ -106,13 +105,34 @@ export default class WatchSidebar {
       name: v.getCode(),
       value: k
     }));
-    WatchesPicker.onConfirmed = item => {
-      this.watchViews[item.value].destroy();
-      this.watchViews.splice(item.value, 1);
-      if (this.watchViews.length === 0) this.addWatch();
-    };
-    WatchesPicker.setItems(watches);
-    WatchesPicker.toggle();
+    watchesPicker = new SelectListView({
+      items: watches,
+      elementForItem: watch => {
+        const element = document.createElement("li");
+        element.textContent = watch.name;
+        element.dataset = watch.name;
+        return element;
+      },
+      didConfirmSelection: watch => {
+        this.watchViews[watch.value].destroy();
+        this.watchViews.splice(watch.value, 1);
```

Comment:
The reason I did this was to avoid accidentally adding null string watches that can't be removed. We could solve this another way on a future pr though.